hal/ia32/console.c

```diff
@@ -3,364 +3,54 @@
-										halconsole_common.attr = (halconsole_common.attr & 0xf0) | ansi2fg[(halconsole_common.parms[i] - 30) & 0x7];
-										break;
-
-									case 40:
-									case 41:
-									case 42:
-									case 43:
-									case 44:
-									case 45:
-									case 46:
-									case 47:
-										halconsole_common.attr = ansi2bg[(halconsole_common.parms[i] - 40) & 0x7] | (halconsole_common.attr & 0x0f);
-										break;
-								}
-							} while (i++ < halconsole_common.parmi);
-
-						default:
-							halconsole_common.esc = esc_init;
-							break;
-					}
-					break;
-
-				case esc_csiqm:
-					switch (c) {
-						case '0':
-						case '1':
-						case '2':
-						case '3':
-						case '4':
-						case '5':
-						case '6':
-						case '7':
-						case '8':
-						case '9':
-							halconsole_common.parms[halconsole_common.parmi] *= 10;
-							halconsole_common.parms[halconsole_common.parmi] += c - '0';
-							break;
-
-						case ';':
-							if (halconsole_common.parmi + 1 < sizeof(halconsole_common.parms))
-								halconsole_common.parmi++;
-							break;
-
-						case 'h':
-							switch (halconsole_common.parms[0]) {
-								case 25:
-									hal_outb(halconsole_common.crtc, 0x0a);
-									hal_outb(halconsole_common.crtc + 1, hal_inb(halconsole_common.crtc + 1) & ~0x20);
-									break;
-							}
-							halconsole_common.esc = esc_init;
-							break;
-
-						case 'l':
-							switch (halconsole_common.parms[0]) {
-								case 25:
-									hal_outb(halconsole_common.crtc, 0x0a);
-									hal_outb(halconsole_common.crtc + 1, hal_inb(halconsole_common.crtc + 1) | 0x20);
-									break;
-							}
-							halconsole_common.esc = esc_init;
-							break;
-
-						default:
-							halconsole_common.esc = esc_init;
-							break;
-					}
-					break;
-			}
-		}
-
-		/* End of line */
-		if (col == halconsole_common.cols) {
-			row++;
-			col = 0;
-		}
-
-		/* Scroll down */
-		if (row == halconsole_common.rows) {
-			i = halconsole_common.cols * (halconsole_common.rows - 1);
-			console_memmove(halconsole_common.vram, halconsole_common.vram + halconsole_common.cols, i);
-			console_memset(halconsole_common.vram + i, (u16)halconsole_common.attr << 8 | ' ', halconsole_common.cols);
-			row--;
-			col = 0;
-		}
-
-		/* Update cursor */
-		i = row * halconsole_common.cols + col;
-		hal_outb(halconsole_common.crtc, 0x0e);
-		hal_outb(halconsole_common.crtc + 1, i >> 8);
-		hal_outb(halconsole_common.crtc, 0x0f);
-		hal_outb(halconsole_common.crtc + 1, i);
-		*((u8 *)(halconsole_common.vram + i) + 1) = halconsole_common.attr;
-	}
-
-	hal_spinlockClear(&halconsole_common.spinlock, &sc);
-}
+#include "hal/console.h"
 
 
 void hal_consolePrint(int attr, const char *s)
 {
-	if (attr == ATTR_BOLD)
-		_hal_consolePrint(CONSOLE_BOLD);
-	else if (attr != ATTR_USER)
-		_hal_consolePrint(CONSOLE_CYAN);
-
-	_hal_consolePrint(s);
-	_hal_consolePrint(CONSOLE_NORMAL);
+#ifdef HAL_CONSOLE_VGA
+	void hal_consoleVGAPrint(int, const char *);
```

Comment:
Consider creating `console-serial.h` and `console-vga.h` to contain these declarations.

Reply:
I created a combined header.